test/lsp.py

```diff
@@ -180,15 +288,332 @@ class Counter:
     failed: int = 0
 
 
-class Marker(Enum):
-    SimpleRange = auto()
-    MultilineRange = auto()
-
-
 # Returns the given marker with the end extended by 'amount'
 def extendEnd(marker, amount=1):
-    marker["end"]["character"] += amount
-    return marker
+    newMarker = deepcopy(marker)
+    newMarker["end"]["character"] += amount
+    return newMarker
+
+class TestParserException(Exception):
+    def __init__(self, incompleteResult, msg: str):
+        self.result = incompleteResult
+        super().__init__("Failed to parse test specification: " + msg)
+
+class TestParser:
+    RequestAndResponse = namedtuple('RequestAndResponse',
+        "method, request, response, responseBegin, responseEnd",
+        defaults=(None, None, None, None)
+    )
+    Diagnostics = namedtuple('Diagnostics', 'tests start end has_header')
+    Diagnostic = namedtuple('Diagnostic', 'marker code')
+
+    TEST_START = "// ----"
```

Comment:
```suggestion
    TEST_START = "\n// ----\n"
```
or similar. should be required to stand at the beginning of a line and the **only** content of that line.